### DISCUSSION TITLE
请问“--el-color-primary-dark-2”有计算规则吗？

### DISCUSSION THREAD

**Discussion by chnljx** [element-plus/element-plus#7152] [Q&A] [answered]
是根据下面的rgb生成的吗？是的话公式是什么？
--el-color-primary: #409EFF; // rgb 64,158,255

**Comment by chnljx** [comment, marked as answer]
@mixin set-color-mix-level(
  $type,
  $number,
  $mode: 'light',
  $mix-color: $color-white
) {
  $colors: map.deep-merge(
    (
      $type: (
        '#{$mode}-#{$number}':
          mix(
            $mix-color,
            map.get($colors, $type, 'base'),
            math.percentage(math.div($number, 10))
          ),
      ),
    ),
    $colors
  ) !global;
}

// --el-color-primary-dark-2
@each $type in $types {
  @include set-color-mix-level($type, 2, 'dark', $color-black);
}

**Reply by chnljx** [reply]
r += Math.round(0.2 * (0 - r)) // 0.2是百分比

**Comment by YunYouJun** [comment]
https://github.com/element-plus/element-plus/blob/bc4f82d7b5a33ebda6a4d8e4652810b3e71a7d53/packages/theme-chalk/src/dark/var.scss#L20-L39

**Reply by chnljx** [reply]
已经解决了，转换成js是：red += Math.round(0.2 * (0 - red)) // 0.2是百分比

**Reply by chnljx** [reply]
按钮的自定义颜色感觉可以按这个算法和sass的算法一致
